### PR TITLE
Added detekt and Fixed some bugs with the criminal fragment

### PR DESCRIPTION
**Pull request by f7deleon**
Added detekt and Fixed some bugs with the criminal fragment

### Related Trello cards:
https://trello.com/c/yAZiLlcL

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -41,11 +40,13 @@ final TIMBER_VERSION = '4.7.0'
 final TEST_ESPRESSO_VERSION = '3.0.2'
 final TEST_RUNNER_VERSION = '1.0.2'
 final JUNIT_VERSION = '4.12'
+final CONSTRAINT_LAYOUT_VERSION = '1.1.2'
```
Comment (mirland) [inline app/build.gradle]: alpha

File: app/build.gradle [hunk 2]
```diff
@@ -41,11 +40,13 @@ final TIMBER_VERSION = '4.7.0'
...
 
 dependencies {
     androidTestImplementation "com.android.support.test.espresso:espresso-core:${TEST_ESPRESSO_VERSION}"
     androidTestImplementation "com.android.support.test:runner:${TEST_RUNNER_VERSION}"
 
+    implementation "com.android.support.constraint:constraint-layout:${CONSTRAINT_LAYOUT_VERSION}"
```
Comment (mirland) [inline app/build.gradle]: is it needed?

File: build.gradle [hunk 3]
```diff
@@ -5,16 +5,19 @@ buildscript {
     repositories {
         google()
         jcenter()
+        maven { url "https://plugins.gradle.org/m2/" }
     }
+
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.3'
+        classpath 'gradle.plugin.io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.0.0.RC2'
```
Comment (mirland) [inline build.gradle]: use the lastest version

File: detekt.gradle [hunk 4]
```diff
@@ -0,0 +1,9 @@
+detekt {
+    version = "1.0.0.RC2"
```
Comment (mirland) [inline detekt.gradle]: same here use the lastest version

**Review by mirland — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (mirland) [reply build.gradle]: this is because they added a lot of improvements

File: app/detekt-config.yml [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    active: false
+
+# *experimental feature*
+# Migration rules can be defined in the same config file or a new one
+migration:
+  active: true
+  imports:
+    # your.package.Class: new.package.or.Class
+    # for example:
+    # io.gitlab.arturbosch.detekt.api.Rule: io.gitlab.arturbosch.detekt.rule.Rule
```
Comment (mirland) [inline app/detekt-config.yml]: nl

**Review by mirland — APPROVED** [review]

File: app/build.gradle [hunk 6]
```diff
@@ -8,10 +9,8 @@ buildscript {
     }
 }
 
-repositories {
-    maven { url 'https://maven.fabric.io/public' }
-}
-
+apply plugin: "io.gitlab.arturbosch.detekt"
+apply from: '../detekt.gradle'
```
Comment (FerrariBruno) [inline app/build.gradle:13]: 😈

**Pull request merged**